habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -164,7 +166,8 @@ def forward(self, observations: Dict[str, torch.Tensor]) -> torch.Tensor:  # typ
             # permute tensor to dimension [BATCH x CHANNEL x HEIGHT X WIDTH]
             rgb_observations = rgb_observations.permute(0, 3, 1, 2)
             rgb_observations = (
-                rgb_observations.float() / 255.0
+                rgb_observations.to(dtype=next(self.parameters()).dtype)
```

Comment:
What about the type of the depth observations? If they remain fp32, then I believe the torch.cat below will convert everything back to fp32.

Reply:
They are already fp16 when in fp16 mode as they stored as fp16 in the rollout storage for further memory savings.

Also, torch.cat won't cast, it'll error if things are different dtypes (unless something changed...)